Remove recommendation to avoid duration > 30 days

dnault opened the pull request:
Since Java SDK 3.0.6, it's safe to specify a duration longer than 30 days.

osfameron commented on modules/howtos/pages/kv-operations.adoc:
```diff
@@ -209,11 +209,7 @@ You can set an expiry value from a `Duration` when creating a document:
-The SDK tries its best to coerce it into sane values, but to avoid any unexpected behavior please stick with relative duration for durations of less than 30 days.
-
-For longer expiry values, use an `Instant` instead:
+Alternatively, you can specify the expiry as an `Instant`:
```
Does that mean that the "magic" behaviour above has been removed?
e.g. any numeric value is a duration, and only an `Instant` object represents a point in time?

dnault replied:
Prior to Java SDK 3.0.6, specifying a duration longer than 30 days would cause the document to expire IMMEDIATELY... but that's no longer the case. Your comment reminded me that there is still some weirdness at the 50 *year* threshold. Updated the PR accordingly.

dnault committed:
Remove recommendation to avoid duration > 30 days

Since Java SDK 3.0.6, it's safe to specify a duration
longer than 30 days. There's still some magic at the
50 *year* threshold though.
